controllers/automation/policyautomation_controller.go

```diff
@@ -199,6 +199,123 @@ func (r *PolicyAutomationReconciler) Reconcile(
 			} else {
 				log.Info("All clusters are compliant. Doing nothing.")
 			}
+		} else if policyAutomation.Spec.Mode == "everyevent" {
+			log := log.WithValues("mode", "everyevent")
+			targetList := common.FindNonCompliantClustersForPolicy(policy)
+			// convert slice targetList to map for search efficiency
+			targetListMap := map[string]bool{}
+			for _, target := range targetList {
+				targetListMap[target] = true
+			}
+			// final target clusters list that the new ansible job will be created on
+			trimmedTargetList := []string{}
+			// delayAfterRunSeconds default value is zero
+			delayAfterRunSeconds := policyAutomation.Spec.DelayAfterRunSeconds
+			log.Info("delayAfterRunSeconds", delayAfterRunSeconds)
+			// automation event time on each cluster
+			eventMap := policyAutomation.Status.ClustersWithEvent
+
+			// original eventTime is earlier than automationStartTime
+			eventTime := time.Now().UTC()
+			eventTimeStr := eventTime.Format(time.RFC3339)
+
+			requeueFlag := false
+
+			for clusterName, clusterEvent := range eventMap {
+				oldAutomationStartTime, err := time.Parse(time.RFC3339, clusterEvent.AutomationStartTime)
+				if err != nil {
+					log.Error(err, "Failed to convert AutomationStartTime in ClustersWithEvent")
+
+					delete(eventMap, clusterName)
+				}
+
+				oldEventTime, err := time.Parse(time.RFC3339, clusterEvent.EventTime)
+				if err != nil {
+					log.Error(err, "Failed to convert EventTime in ClustersWithEvent")
+
+					delete(eventMap, clusterName)
+				}
+
+				delayedTime := oldAutomationStartTime.Add(
+					time.Duration(delayAfterRunSeconds) * time.Second)
```

Comment:
Here there is a complaint since delayAfterRunSeconds is unit and Second is time.Duration type, my original meaning is converted delayAfterRunSeconds to time.Duration type first then do multiply with Second.